### PR TITLE
[SYCLomatic] Add workaround for templated kernel call in macro

### PR DESCRIPTION
**Pull request by AndyCHHuang**

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/lib/DPCT/AnalysisInfo.cpp [hunk 1]
```diff
@@ -2825,6 +2825,7 @@ std::string CallFunctionExpr::getTemplateArguments(bool &IsNeedWarning,
     if ((TA.isNull() || !TA.isWritten()) && WrittenArgsOnly)
       continue;
     std::string Str = TA.getString();
+
```
Comment (zhimingwang36) [inline clang/lib/DPCT/AnalysisInfo.cpp]: there is no logic change here, to keep small footprint for this PR, so please revert this change.

**Review by zhimingwang36 — APPROVED** [review]
LGTM
